lib/emerald/parser.rb

```diff
@@ -23,7 +23,9 @@ def parse_node(source)
       when " "
         parse_whitespace(source)
       when /[a-zA-Z\-]/
-        parse_atom(source)
```

Comment:
I'd probably keep the parser functions always taking the source as an arg, they can pull the first character off themselves if they want to.